bin/imsim.py

```diff
@@ -0,0 +1,120 @@
+for the DESC collaboration and LSST project.  This version of the program can
+read phoSim instance files as is. It leverages the LSST Sims GalSim interface
+code found in sims_GalSimInterface.
+"""
+from __future__ import absolute_import, print_function
+import os
+import sys
+import logging
+import argparse
+
+from lsst.obs.lsstSim import LsstSimMapper
+
+from lsst.sims.photUtils import LSSTdefaults
+from lsst.sims.utils import ObservationMetaData
+
+from lsst.sims.GalSimInterface.galSimCatalogs import GalSimBase
+from lsst.sims.GalSimInterface import GalSimStars
+from lsst.sims.GalSimInterface import GalSimGalaxies
+
+import desc.imsim
+from desc.imsim.monkeyPatchedGalSimBase import \
+    phoSimCalculateGalSimSeds, phoSimInitilizer, get_phoSimInstanceCatalog
+
+def main(argv):
+    """
+    Drive GalSim to simulate the LSST.
+    """
+    # Monkey Patch the GalSimBase class and replace the routines that
+    # gets the objects and builds the catalog with my version that gets the
+    # information from a phoSim instance catalog file.  This should be temporary
+    # and is a hack.
+    GalSimBase.__init__ = phoSimInitilizer
+    GalSimBase._calculateGalSimSeds = phoSimCalculateGalSimSeds
+    GalSimBase.get_fitsFiles = get_phoSimInstanceCatalog
+
+    # Setup logging output.
+    logging.basicConfig(format="%(message)s", level=logging.INFO,
+                        stream=sys.stdout)
+    logger = logging.getLogger()
+
+    # Setup a parser to take command line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s 0.0')
+    parser.add_argument('-f', '--file', help="Specify the instance file")
+    parser.add_argument('-n', '--numrows', default=2000, type=int,
+                        help="read the first numrows of the file.")
+    parser.add_argument('--outdir', type=str, default='fits',
+                        help='output directory for eimage file')
+    arguments = parser.parse_args()
+
+    # Get the number of rows to read from the instance file.  Use
+    # default if not specified.
+    numRows = arguments.numrows
+    logger.info("Reading %i rows from the instance catalog %s.",
+                numRows, arguments.file)
+
+    # The PhoSim instance file contains both pointing commands and
+    # objects.  The parser will split them and return a both phosim
+    # command dictionary and a dataframe of objects.
+    commandDictionary, phoSimObjectList = \
+        desc.imsim.parsePhoSimInstanceFile(arguments.file, numRows)[:2]
+
+    # Access the relevant pointing commands from the dictionary.
+    visitIDString = str(int(commandDictionary['obshistid'][0]))
+    mjd = commandDictionary['mjd'][0]
+    rightAscension = commandDictionary['rightascension'][0]
+    declination = commandDictionary['declination'][0]
+    rotSkyPosition = commandDictionary['rotskypos'][0]
+    bandpass = 'ugrizy'[int(commandDictionary['filter'][0])]
```

Comment:
Unfortunately I can't figure out how to comment on code that wasn't actually changed but notice in monkeyPatchedGalSimBase.py this code:


    # Add bandpasses to simulate over.
    self.bandpassNames = ['r']
    # self.bandpassNames = self.obs_metadata.bandpassName

This means no matter what currently the r band will be used.  We need to tie it to the bandpass passed through the observationmetadata.